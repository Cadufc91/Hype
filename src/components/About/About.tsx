import * as React from 'react';
import {
    AboutContainer,
    AboutImages,
    ComputerImg,
    WhiteBg,
    InfoContainer,
    InfoHeading,
    InfoText,
    Experience,
    ContactBtn
} from './style';
import Computer from '../../assets/img/about.png';
import { BsArrowRight } from 'react-icons/bs';

const About = () => {
    return(
        <AboutContainer id='about'>
            <AboutImages>
                <ComputerImg src={Computer} alt="" />
                <WhiteBg />
            </AboutImages>
            <InfoContainer>
                <InfoHeading>Software Engineer and Infrastructure specialist based in Brazil.</InfoHeading>
                <InfoText>Morbi quam velit, euismod in imperdiet vitae, elementum et elit. Nunc finibus, felis sit amet sollicitudin sollicitudin, nisi magna feugiat enim, in maximus urna enim ac tortor. Nunc in volutpat ipsum, molestie commodo odio. Quisque auctor nisi mi. Aenean venenatis sapien et interdum interdum.</InfoText>
                <Experience>
                    <div>
                        <h3>17+ </h3><span>Years of <br></br>Experience</span>
                    </div>
                    <div>
                        <h3>325+ </h3><span>Completed <br></br>Projects</span>
                    </div>
                </Experience>
                <ContactBtn>Say HI <BsArrowRight size={30} /></ContactBtn>
            </InfoContainer>
        </AboutContainer>
    )
};

export default About;