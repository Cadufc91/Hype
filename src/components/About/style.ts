import styled from "styled-components";

export const AboutContainer = styled.div`
    display: flex;
    flex-direction: row;
    margin-top: 300px;
    height: 795px;
`;

export const AboutImages = styled.div`
    margin-left: 15%;
    margin-top: 100px;
    max-width: 30%;
`;

export const ComputerImg = styled.img`
    width: 100%;
`;

export const InfoContainer = styled.div`
    margin-left: 80px;
    margin-right: 5%;
    width: 672px;
    margin-top: 70px;
`;

export const InfoHeading = styled.h2`
    color: #fff;
    font-size: 48px;
    font-weight: 500;
    margin-bottom: 20px;
`;

export const InfoText = styled.p`
    font-size: 18px;
    font-weight: 400;
    color: #A5A5A5;
    `;
    
export const Experience = styled.div`
    display: flex;
    flex-direction: row;

    div {
        display: flex;
        flex-direction: row;
        align-items: center;

        h3 {
            font-size: 40px;
            font-weight: 400;
            color: #ED1CA6;
            margin-right: 16px;
        }

        span {
           
            font-size: 18px;
            color: #fff;
            margin-right: 50px;
        }
    }
`;

export const ContactBtn = styled.button`
    width: 162px;
    height: 56px;
    font-size: 16px;
    font-weight: 600;
    color: #fff;
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 0 32px;
    text-transform: uppercase;
    background: #7D00FF;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    :hover{
        transform: translateY(-10px);
        opacity: 0.5;
    }
`;