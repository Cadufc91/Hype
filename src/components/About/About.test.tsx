import { render } from "@testing-library/react";
import  { About } from './About';

describe('Header', () => {
    it('should match snapshot', () => {
        const { container } = render(<About />)

        expect(container.firstChild).toMatchSnapshot()
    })
})