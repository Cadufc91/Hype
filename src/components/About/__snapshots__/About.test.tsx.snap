// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div
  class="sc-bczRLJ bYqgBn"
  id="about"
>
  <div
    class="sc-gsnTZi kEZgUy"
  >
    <img
      alt=""
      class="sc-dkzDqf bxnkfQ"
      src="about1.png"
    />
  </div>
  <div
    class="sc-hKMtZM bBbXrc"
  >
    <h2
      class="sc-eCYdqJ jJXZxN"
    >
      Desenvolvendo sites e aplicativos com a melhor experiência do usuário.
    </h2>
    <p
      class="sc-jSMfEi kqVGmU"
    >
      Minhas experiências anteriores trazems uma visão inovadora de quem já esteve do outro lado do desenvolvimento de sites/aplicativos podendo assim contribuir para uma melhor experiência de usuário e desenvolvimento de projetos para todos.
    </p>
    <div
      class="sc-gKXOVf blkNSA"
    >
      <div>
        <h3>
          1 
        </h3>
        <span>
          Ano de 
          <br />
          Experiência
        </span>
      </div>
      <div>
        <h3>
          20+ 
        </h3>
        <span>
          Projetos 
          <br />
          Concluídos
        </span>
      </div>
    </div>
    <button
      class="sc-iBkjds iATNfq"
    >
      <a
        href="mailto:fernandes.cadu@gmail.com"
      >
        Diga oi
      </a>
       
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
